### PR TITLE
Fix chunk::make for buffers with SBO

### PR DESCRIPTION
**Pull request by dominiklohmann**
### :notebook_with_decorative_cover: Description

See comment in diff.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Follow the logic, trust our unit test suite. Or try to come up with something better.

### CODE REVIEW

File: libvast/vast/chunk.hpp [hunk 1]
```diff
@@ -81,8 +81,12 @@ class chunk final : public caf::ref_counted {
     requires(!(std::is_lvalue_reference_v<
                  Buffer> || std::is_trivially_move_assignable_v<Buffer>))
   static auto make(Buffer&& buffer) -> decltype(as_bytes(buffer), chunk_ptr{}) {
```
Comment (lava) [inline libvast/vast/chunk.hpp:83]: Btw, is there a special reason why we kept the `as_bytes(buffer)` as SFINAE and didn't move it into the `requires` clause?

Comment (dominiklohmann) [reply libvast/vast/chunk.hpp]: I've pushed the change.

File: libvast/vast/chunk.hpp [hunk 2]
```diff
@@ -78,11 +78,18 @@ class chunk final : public caf::ref_counted {
   /// accidental copies when calling this function.
   /// @returns A chunk pointer or `nullptr` on failure.
   template <class Buffer>
-    requires(!(std::is_lvalue_reference_v<
-                 Buffer> || std::is_trivially_move_assignable_v<Buffer>))
-  static auto make(Buffer&& buffer) -> decltype(as_bytes(buffer), chunk_ptr{}) {
-    const auto view = as_bytes(buffer);
-    return make(view, [buffer = std::exchange(buffer, {})]() noexcept {
+    requires(!std::is_lvalue_reference_v<Buffer> &&          //
+             !std::is_trivially_move_assignable_v<Buffer> && //
```
Comment (lava) [inline libvast/vast/chunk.hpp:82]: I think we can drop the `!is_trivially_move_assignable` requirement now, right?

File: libvast/vast/chunk.hpp [hunk 3]
```diff
@@ -78,11 +78,18 @@ class chunk final : public caf::ref_counted {
...
+             requires(const Buffer& buffer) {
+               { as_bytes(buffer) } -> concepts::same_as<view_type>;
+             })
+  static auto make(Buffer&& buffer) -> chunk_ptr {
+    // Move the buffer into a unique pointer; otherwise, we might run into
+    // issues with small buffer optimizations where the view no longer points
+    // to the correct data once we moved the buffer into the deleter.
```
Comment (lava) [inline libvast/vast/chunk.hpp:89]: The comment seems to be too narrow, even without small buffer optimization there is no requirement that a move constructor must preserve all internal memory addresses.

**Review by lava — APPROVED** [review]
Looks good to me, just some comments on tangential issues.

File: libvast/vast/chunk.hpp (hunk 3 above)
Comment (dominiklohmann) [reply libvast/vast/chunk.hpp]: I reworded the comment in a separate commit.

File: libvast/vast/chunk.hpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/vast/chunk.hpp]: Good catch. I dropped the requirement in a separate commit.

**Pull request merged**